### ISSUE TITLE
Allow "Maybe" to be exported, so it can be imported into a implemented class

### ISSUE THREAD

**Issue by appsolutegeek** [dotansimha/graphql-code-generator#1676]
I am using  "implements" on the defined types, this is so I have better control over mapping using class-transformer (another library) which allows annotating over fields.

currently, it is 

```
type Maybe<T> = T | undefined;
```

So when I implement a type, the fields using Maybe<type> .. but I don't have access to Maybe as its not exported.

Would be nice to have a config setting to enable exporting or not, just as it has the types that it should use i.e. undefined | null . etc

**Comment by dotansimha** [comment]
Hi @appsolutegeek ,
Please use this configuration: https://graphql-code-generator.com/docs/plugins/typescript#maybevalue-string-default-value-t-null

```yml
generates:
   path/to/file.ts:
     plugins:
        - typescript
      config:
          maybeValue: T | undefined
```

**Comment by appsolutegeek** [comment]
Thanks @dotansimha, i am already using that config.. But the type isn't being exported i.e. its

```
type Maybe<T> = T | undefined;
```

and NOT

```
export type Maybe<T> = T | undefined;

```

All over types are being exported but Maybe is not..

**Comment by dotansimha** [comment]
@appsolutegeek you are right, I thought it does: https://github.com/dotansimha/graphql-code-generator/blob/master/packages/plugins/typescript/src/index.ts#L104

**Comment by kamilkisiela** [comment]
#1718

**Comment by dotansimha** [comment]
Fixed in `1.1.0` 🚀
